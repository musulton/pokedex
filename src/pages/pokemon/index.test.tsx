import {render} from "@testing-library/react";

import Pokemon from "@/pages/pokemon/index";
import React, {useState} from "react";

jest
    .mock("../../components/ImageCard", () => "ImageCard")
    .mock("../../services/api", () => ({
        fetchData: jest.fn()
    }))
    .mock("../../utils/common", () => ({
        ...jest.requireActual("../../utils/common"),
        getRandomColor: jest.fn().mockImplementation(() => "white")
    }))
    .mock('react', () => ({
        ...jest.requireActual('react'),
        useState: jest.fn(),
    }))

describe("Pokemon pages", () => {
    const mockState = {
        count: 100,
        pokemon: [{
            id: 1,
            name: "Pokemon",
            imageSrc: "http://gugel.com"
        }]
    }

    beforeEach(() => {
        useState.mockImplementation(() => [mockState, jest.fn()])
    })

    it("should render Pokemon correctly", () => {
        const {container} = render(<Pokemon />)

        expect(container).toMatchSnapshot();
    })
})
