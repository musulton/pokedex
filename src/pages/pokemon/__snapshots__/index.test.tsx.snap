// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon pages should render Pokemon correctly 1`] = `
<main
  class="sc-85dca682-0 eZnztt"
>
  <div
    class="sc-85dca682-5 gGzqsq"
  >
    <h1
      class="sc-85dca682-3 ipXhU"
    >
      Pokémon
    </h1>
    <p
      class="sc-85dca682-4 iCeUpT"
    >
      Pokémon are mysterious creatures with many secrets. Some Pokémon live with humans and some live in the wild in meadows, caves, or the sea.
    </p>
  </div>
  <div
    class="sc-85dca682-2 ddhbNf"
  >
    <imagecard
      imageitem="[object Object]"
    />
  </div>
  <div
    class="paginationWrapper"
  >
    <button
      class="paginationButton"
    >
      ‹ Previous
    </button>
     
    <button
      class="paginationButton"
    >
      Next ›
    </button>
  </div>
</main>
`;
